src/main/java/VASSAL/build/module/map/ForwardToKeyBuffer.java

```diff
@@ -87,10 +88,10 @@ protected void process(KeyEvent e) {
       lastConsumedEvent = null;
     }
     final int c = e.getKeyCode();
-    // Don't pass SHIFT or CONTROL only to counters
-    if (!e.isConsumed() && c != KeyEvent.VK_SHIFT && c != KeyEvent.VK_CONTROL) {
+    // Don't pass modifier keys alone to counters
+    if (!e.isConsumed() && c != KeyEvent.VK_SHIFT && c != KeyEvent.VK_CONTROL && c != KeyEvent.VK_ALT && c != KeyEvent.VK_META) {
```

Comment:
For such complex conditions, I like to create a boolean variable and use the text I would otherwise put in the comment for this variable's name, e.g.:
```
final boolean onlyModifierKeys = (c == KeyEvent.VK_SHIFT || c == KeyEvent.VK_CONTROL || c == KeyEvent.VK_ALT || c == KeyEvent.VK_META);
if (!e.isConsumed() && !onlyModifierKeys) {
...
```

This makes the `if` easier to read by making the comment part of the code.

Reply:
By the way this is one reason why I personally prefer to see all the (c != KeyEvent.VK_SHIFT) things individually in parentheses. I know they aren't formally necessary but boy do they make the if statements more readable. 

Anyway, did what you suggested.

Reply:
I agree, that is my thought as well. Just because you can write the code in fewer characters doesn't mean you should. It makes no difference at all to a modern compiler, but makes the visual parsing of long if statements FAR easier.

Reply:
That line is just too long for easy reading. Splitting the conditions across two or more lines would help.

Reply:
Already done and checked in